samples/kinesis_file_gstreamer_sample_app.cpp

```diff
@@ -0,0 +1,797 @@
+#include <gst/gst.h>
+#include <gst/app/gstappsink.h>
+#include <string.h>
+#include <chrono>
+#include <Logger.h>
+#include "KinesisVideoProducer.h"
+#include <vector>
+#include <stdlib.h>
+#include <mutex>
+#include <IotCertCredentialProvider.h>
+
+using namespace std;
+using namespace com::amazonaws::kinesis::video;
+using namespace log4cplus;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int gstreamer_init(int, char **);
+
+#ifdef __cplusplus
+}
+#endif
+
+LOGGER_TAG("com.amazonaws.kinesis.video.gstreamer");
+
+#define DEFAULT_RETENTION_PERIOD_HOURS 2
```

Comment:
Once we are moving all of these to the samples repository, I would highly recommend having a Common directory which has a set of sources and definitions which will be shared among the samples. In this case, all of these definitions will go to Common.h or something

Reply:
Yep, that makes sense, similar to what we have in webrtc sdk